src/node_sqlite.cc

```diff
@@ -128,6 +148,194 @@ inline void THROW_ERR_SQLITE_ERROR(Isolate* isolate, int errcode) {
   isolate->ThrowException(error);
 }
 
+class BackupJob : public ThreadPoolWork {
+ public:
+  explicit BackupJob(Environment* env,
+                     DatabaseSync* source,
+                     Local<Promise::Resolver> resolver,
+                     std::string source_db,
+                     std::string destination_name,
+                     std::string dest_db,
+                     int pages,
+                     Local<Function> progressFunc)
+      : ThreadPoolWork(env, "node_sqlite3.BackupJob"),
+        env_(env),
+        source_(source),
+        pages_(pages),
+        source_db_(source_db),
+        destination_name_(destination_name),
+        dest_db_(dest_db) {
+    resolver_.Reset(env->isolate(), resolver);
+    progressFunc_.Reset(env->isolate(), progressFunc);
+  }
+
+  void ScheduleBackup() {
+    Isolate* isolate = env()->isolate();
+    HandleScope handle_scope(isolate);
+
+    backup_status_ = sqlite3_open(destination_name_.c_str(), &pDest_);
+
+    Local<Promise::Resolver> resolver =
+        Local<Promise::Resolver>::New(env()->isolate(), resolver_);
+
+    Local<Object> e = Local<Object>();
+
+    if (backup_status_ != SQLITE_OK) {
+      if (!CreateSQLiteError(isolate, pDest_).ToLocal(&e)) {
+        Finalize();
+
+        return;
+      }
+
+      Finalize();
```

Comment:
There are an awful lot of `Finalize()` calls scattered around the code. Is it possible to centralize at least some of them?

Reply:
Yeah, that can be better. So, what if we I remove those from if conditions like

```cpp
if (!CreateSQLiteError(isolate, pDest_).ToLocal(&e)) {
  Finalize(); // <- remove this
        
  return;
}
```

My assumption, which can be wrong, is that if there's an error and we can't create an error object, we have bigger problems.

Reply:
I think we can have a method for handling when an error happens. So it centralizes error handling in there.

Reply:
Done. Added a dedicated method for that.